src/gt4py/analysis/passes.py

```diff
@@ -1210,6 +1210,95 @@ def apply(self, transform_data: TransformData) -> TransformData:
         return transform_data
 
 
+class ReduceTemporaryStoragesPass(TransformPass):
+    class StorageReducer(gt_ir.IRNodeMapper):
+        def __init__(self):
+            self.iir = None
+            self.interval = None
+            self.iteration_order = None
+            self.full_fields = set()
+            self.reduced_fields = dict()
+            self.fields_extents = dict()
+
+        def __call__(self, node: gt_ir.StencilImplementation) -> gt_ir.StencilImplementation:
+            assert isinstance(node, gt_ir.StencilImplementation)
+            result = self.visit(node)
+            self._reduce_fields()
+            return result
+
+        def _reduce_fields(self):
```

Comment:
This could be done more nicely in a separate Vistior. (i.e. the `ReduceTemporaryStoragesPass` could contain two Visitors, e.g.  `Detect2DTemporaryVisitor` and `StorageReducer` or something.